src/Framework/PaymentGateways/PaymentGateway.php

```diff
@@ -78,17 +77,17 @@ public function supportsSubscriptions()
      *
      * @inheritDoc
      */
-    public function handleCreatePayment(GatewayPaymentData $gatewayPaymentData)
+    public function handleCreatePayment(Donation $donation)
```

Comment:
@ravinderk this is clearly a drastic _breaking_ change to the api replacing `GatewayPaymentData` with the `Donation` model.  Combing through these changes it _appears_ the model will give gateways everything they need to function properly so I _do_ prefer this.  If we agree to this change we will need to make sure we are documenting this properly as this could break existing gateways using the api.  

I'm still looking over the changes but wanted to start this discussion early as it will have a big impact.